src/BuiltInTools/DotNetDeltaApplier/HotReloadAgent.cs

```diff
@@ -82,24 +82,36 @@ private UpdateHandlerActions GetMetadataUpdateHandlerActions()
             var handlerActions = new UpdateHandlerActions();
             foreach (var assembly in sortedAssemblies)
             {
-                foreach (var attr in assembly.GetCustomAttributesData())
+                try
                 {
-                    // Look up the attribute by name rather than by type. This would allow netstandard targeting libraries to
-                    // define their own copy without having to cross-compile.
-                    if (attr.AttributeType.FullName != "System.Reflection.Metadata.MetadataUpdateHandlerAttribute")
+                    foreach (var attr in assembly.GetCustomAttributesData())
                     {
-                        continue;
+                        // Look up the attribute by name rather than by type. This would allow netstandard targeting libraries to
+                        // define their own copy without having to cross-compile.
+                        if (attr.AttributeType.FullName != "System.Reflection.Metadata.MetadataUpdateHandlerAttribute")
+                        {
+                            continue;
+                        }
+
+                        IList<CustomAttributeTypedArgument> ctorArgs = attr.ConstructorArguments;
+                        if (ctorArgs.Count != 1 ||
+                            ctorArgs[0].Value is not Type handlerType)
+                        {
+                            _log($"'{attr}' found with invalid arguments.");
+                            continue;
+                        }
+
+                        GetHandlerActions(handlerActions, handlerType);
                     }
-
-                    IList<CustomAttributeTypedArgument> ctorArgs = attr.ConstructorArguments;
-                    if (ctorArgs.Count != 1 ||
-                        ctorArgs[0].Value is not Type handlerType)
-                    {
-                        _log($"'{attr}' found with invalid arguments.");
-                        continue;
-                    }
-
-                    GetHandlerActions(handlerActions, handlerType);
+                }
+                catch (Exception e)
```

Comment:
It'd be better to scope the try-catch just around `GetCustomAttributesData()` instead of the entire loop. That way, if we have a bug in the code that throws we'll get a better stack trace.

Reply:
Interesting. I'll make the change.

Reply:
Moved to a separate method.